src/adapters/Adapter.ts

```diff
@@ -0,0 +1,275 @@
+        this.profile.id = user.id;
+        this.bot.say(`[${new Date().toLocaleString()}] Login Successfully!`);
+    }
+
+    async logoutHandler(user: ContactSelfInterface | Contact): Promise<void> {
+        return logoutAccount(user, this.profile.source);
+    }
+
+    async messageHandler(message: MessageInterface): Promise<void> {
+        this.logger.silly(
+            `${message.type()} Message received, from ${message.talker()} to ${message.listener()}, content: ${message.text()}`
+        );
+        const self = message.talker().self() && message.listener()?.self();
+        if (self) {
+            switch (message.type()) {
+                case MessageType.Text:
+                    const str = message.text();
+                    const [cmd, ...args] = str.split(" ");
+                    this.invokeCommand(cmd, ...args);
+                    return;
+                case MessageType.Contact:
+                    // Forward contact to yourself to set messages aggregation target accout.
+                    const contact = await message.toContact();
+                    await this.invokeCommand("setfta", contact);
+                    return;
+                default:
+                    break;
+            }
+        }
+        if (!this.filter(message)) {
+            this.logger.info(
+                `Message from ${message.talker()} has been filtered: ${message.text()}`
+            );
+            return;
+        }
+        this.emit("message", message);
+        this.saveMessage(message).catch(this.logger.error);
+    }
+
+    private async heatBeatHandler(data: any): Promise<void> {
+        this.logger.silly(data);
+    }
+
+    private async errorHandler(error: Error): Promise<void> {
+        this.logger.error(error);
+    }
+
+    async convertMessagesToSayable(messages: Message[]): Promise<Sayable[]> {
+        throw new Error("Not implemented");
+    }
+
+    async saveMessage(message: MessageInterface): Promise<void> {
+        throw new Error("Not implemented");
+    }
+
+    public loadExtension(ext: Extension) {
+        if (this._extensions[ext.name]) {
+            throw new Error(`Extension ${ext.name} already exists.`);
+        }
+        this._extensions[ext.name] = ext;
+        ext.register();
+        this.logger.silly(`Extension ${ext.name} loaded.`);
+    }
+
+    public unregisertExtension(ext: Extension) {
+        delete this._extensions[ext.name];
+        this.logger.silly(`Extension ${ext.name} unregistered.`);
+    }
+
+    public registerCommand(cmd: Command) {
+        // shortcut should be unique
+        if (this._commands.hasOwnProperty(cmd.shortcut)) {
+            throw new Error(`Command ${cmd.shortcut} already exists`);
+        }
+        this._commands[cmd.shortcut] = cmd;
+        this.logger.silly(`Command ${cmd.name} registered.`);
+    }
+
+    public async invokeCommand(shortcut: string, ...args: any[]) {
+        if (this._commands[shortcut]) {
+            this.logger.info(
```

Comment:
Why start a newline here?

Reply:
This line is too long, so my prettier plugin has formatted it by start a new line.